status/reporters.go

```diff
@@ -172,37 +226,185 @@ func (r *GitHubPipelineRunReporter) createCheckRunAdapter() (*github.CheckRunAda
+		description = scenario + " has started"
+	} else {
+		outcome, err := helpers.CalculateIntegrationPipelineRunOutcome(r.logger, pipelineRun)
+		if err != nil {
+			return err
+		}
+
+		if outcome {
+			state = "success"
+			description = scenario + " has succeeded"
+		} else {
+			state = "failure"
+			description = scenario + " has failed"
+		}
+	}
+
+	_, err := r.client.CreateCommitStatus(ctx, owner, repo, SHA, state, description, statusContext)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *GitHubReporter) createComment(ctx context.Context, pipelineRun *tektonv1beta1.PipelineRun) error {
+	labels := pipelineRun.GetLabels()
+
+	succeeded := pipelineRun.Status.GetCondition(apis.ConditionSucceeded)
+	if succeeded.IsUnknown() {
+		return nil
+	}
+
+	scenario, found := labels[gitops.SnapshotTestScenarioLabel]
+	if !found {
+		return fmt.Errorf("PipelineRun label not found %q", gitops.SnapshotTestScenarioLabel)
+	}
+
+	owner, found := labels[gitops.PipelineAsCodeURLOrgLabel]
+	if !found {
+		return fmt.Errorf("PipelineRun label not found %q", gitops.PipelineAsCodeURLOrgLabel)
+	}
+
+	repo, found := labels[gitops.PipelineAsCodeURLRepositoryLabel]
+	if !found {
+		return fmt.Errorf("PipelineRun label not found %q", gitops.PipelineAsCodeURLRepositoryLabel)
+	}
+
+	issueNumberStr, found := pipelineRun.GetAnnotations()[gitops.PipelineAsCodePullRequestAnnotation]
+	if !found {
+		return fmt.Errorf("PipelineRun label not found %q", gitops.PipelineAsCodeURLRepositoryLabel)
+	}
+
+	issueNumber, err := strconv.Atoi(issueNumberStr)
+	if err != nil {
+		return err
+	}
+
+	results, err := helpers.GetHACBSTestResultsFromPipelineRun(r.logger, pipelineRun)
+	if err != nil {
+		return err
+	}
+
+	outcome, err := helpers.CalculateIntegrationPipelineRunOutcome(r.logger, pipelineRun)
+	if err != nil {
+		return err
+	}
+
+	var title string
+	if outcome {
+		title = scenario + " has succeeded"
+	} else {
+		title = scenario + " has failed"
+	}
+
+	comment, err := FormatComment(title, results)
+	if err != nil {
+		return err
+	}
+
+	_, err = r.client.CreateComment(ctx, owner, repo, issueNumber, comment)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
 // ReportStatus creates/updates CheckRuns when using GitHub App integration.
-func (r *GitHubPipelineRunReporter) ReportStatus(ctx context.Context) error {
-	if !helpers.HasLabelWithValue(r.pipelineRun, gitops.PipelineAsCodeEventTypeLabel, gitops.PipelineAsCodePullRequestType) {
+// When using GitHub webhook integration a commit status and, in some cases, a comment is created.
+func (r *GitHubReporter) ReportStatus(ctx context.Context, pipelineRun *tektonv1beta1.PipelineRun) error {
+	if !helpers.HasLabelWithValue(pipelineRun, gitops.PipelineAsCodeEventTypeLabel, gitops.PipelineAsCodePullRequestType) {
 		return nil
 	}
 
-	if r.installationID > 0 {
-		checkRun, err := r.createCheckRunAdapter()
+	if helpers.HasAnnotation(pipelineRun, gitops.PipelineAsCodeInstallationIDAnnotation) {
```

Comment:
minor suggestion: adding a comment explaining that the `gitops.PipelineAsCodeInstallationIDAnnotation` is used to identify that the GitHub App integration is in place for the PaC pipelineRun.

Reply:
Good point. Comment added.

Reply:
Thanks!